### PR TITLE
Support for Extended Query in pgsql

### PR DESCRIPTION
**Pull request by MingchunZhao**
(issue #78)Support for Extended Query in pgsql

### CODE REVIEW

File: packetbeat/protos/pgsql/parse.go [hunk 1]
```diff
@@ -265,6 +271,53 @@ func (pgsql *Pgsql) parseErrorResponse(s *PgsqlStream, length int) (bool, bool)
 	return true, true
 }
 
+func (pgsql *Pgsql) parseParse(s *PgsqlStream, length int) (bool, bool) {
```
Comment (urso) [inline packetbeat/protos/pgsql/parse.go]: maybe find a better name? parseParse sounds like the most top-level parser function. What's type `'p'` stand for?

File: packetbeat/protos/pgsql/parse.go [hunk 2]
```diff
@@ -265,6 +271,53 @@ func (pgsql *Pgsql) parseErrorResponse(s *PgsqlStream, length int) (bool, bool)
...
+
+	// Ignore SET statement
+	if strings.HasPrefix(m.Query, "SET ") {
+		m.toExport = false
+	}
+	s.parseState = PgsqlExtendedQueryState
+	return pgsql.parseMessageExtendedQuery(s)
+}
+
+func (pgsql *Pgsql) parseParseCompletion(s *PgsqlStream, length int) (bool, bool) {
```
Comment (urso) [inline packetbeat/protos/pgsql/parse.go]: maybe call it `parseExtCompl` or 'parseExtResp` to make clear this method is about handling extended queries?

**Comment by urso** [comment]
Thanks for updated PR. Changes look good, maybe some naming updates to make ext message parsing stand out a little more.

Did you sign the CLA already?.

File: packetbeat/protos/pgsql/parse.go (hunk 1 above)
Comment (MingchunZhao) [inline packetbeat/protos/pgsql/parse.go]: 'P' means 'Parse' process in an extended query.

File: packetbeat/protos/pgsql/parse.go (hunk 2 above)
Comment (MingchunZhao) [inline packetbeat/protos/pgsql/parse.go]: Ok, thanks!

**Comment by MingchunZhao** [comment]
@urso, Yes, I signed the CLA and recieved an email titled:
'Individual Contributor License Agreement under CCLA between elasticsearch BV and Mingchun Zhao is Signed and Filed!'.

**Comment by urso** [comment]
OK, thanks. Have to check why CLA is failing :(

**Comment by tsg** [comment]
We're good with the CLA, so I just merged it. Many thanks for your contribution.

**Comment by MingchunZhao** [comment]
@tsg, @urso, Thanks for your help!

**Pull request merged**